### PR TITLE
allow devel builds of go to pass make version check

### PR DESCRIPTION
**Pull request by gtank**
I'm sure I'm not the only one who runs these. It's nice to be able to stay on one go version instead of switching for different projects.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins, test this please

File: Makefile [hunk 1]
```diff
@@ -17,8 +17,8 @@ GOOSES = darwin linux
 NOTARY_BUILDTAGS ?= pkcs11
 NOTARYDIR := /go/src/github.com/docker/notary
 
-GO_VERSION := $(shell go version | grep "1\.[6-9]\(\.[0-9]+\)*")
-# check to make sure we have the right version
+GO_VERSION := $(shell go version | grep "1\.[6-9]\(\.[0-9]+\)*\|devel")
+# check to make sure we have the right version. developers are on their own.
```
Comment (cyli) [inline Makefile]: Apologies for the nitpick, but would you mind clarifying this statement to say something like "development versions of Go are not officially supported, but allowed for building"?

**Comment by cyli** [comment]
Thank you so much for adding support for go development versions, @gtank!  Sorry for the delay - this LGTM with the above clarification!

File: Makefile (hunk 1 above)
Comment (gtank) [inline Makefile]: No problem, that's better.

**Comment by riyazdf** [comment]
LGTM!

**Comment by cyli** [comment]
jenkins, test this please

**Comment by cyli** [comment]
Jenkins, test this please

**Pull request merged**